packages/safe-core-sdk-types/src/types.ts

```diff
@@ -252,3 +252,51 @@ export interface MetaTransactionOptions {
   gasToken?: string
   isSponsored?: boolean
 }
+
+export type UserOperation = {
+  sender: string
+  nonce: string
+  initCode: string
+  callData: string
+  callGasLimit: bigint
+  verificationGasLimit: bigint
+  preVerificationGas: bigint
+  maxFeePerGas: bigint
+  maxPriorityFeePerGas: bigint
+  paymasterAndData: string
+  signature: string
+}
+
+export type SafeUserOperation = {
+  safe: string
+  nonce: bigint
+  initCode: string
+  callData: string
+  callGasLimit: bigint
+  verificationGasLimit: bigint
+  preVerificationGas: bigint
+  maxFeePerGas: bigint
+  maxPriorityFeePerGas: bigint
+  paymasterAndData: string
+  validAfter: number
+  validUntil: number
+  entryPoint: string
+}
+
+export type EstimateGasData = {
+  maxFeePerGas?: bigint
+  maxPriorityFeePerGas?: bigint
+  preVerificationGas?: bigint
+  verificationGasLimit?: bigint
+  callGasLimit?: bigint
+}
+
+export interface SafeOperation {
```

Comment:
I know all things are package-scoped but I it would be cool if we could avoid having things with the same name. Some developers will do a deep search and find two things and will not be sure which one is which.

Reply:
Good point! I just wonder which name would make sense then. Because the purpose of the interface here is to allow importing the type from `safe-core-sdk-types` instead of from the relay-kit, where it's implemented. 
For consistency, maybe we could rename the [SafeOperation class](https://github.com/safe-global/safe-core-sdk/blob/feat/4337-api-integration/packages/relay-kit/src/packs/safe-4337/SafeOperation.ts) to `EthSafeOperation` so it would be the same pattern as e.g. for [EthSafeTransaction](https://github.com/safe-global/safe-core-sdk/blob/feat/4337-api-integration/packages/protocol-kit/src/utils/transactions/SafeTransaction.ts).

Reply:
One typical pattern is add the "I" letter as prefix so SafeOperation implements ISafeOperation

Reply:
Thought about this already, but not a big fan of it. Also found this in the [TS style guide](https://ts.dev/style/#naming-style):
> Do not mark interfaces specially (IMyInterface or MyFooInterface) unless it's idiomatic in its environment. When introducing an interface for a class, give it a name that expresses why the interface exists in the first place (e.g. class TodoItem and interface TodoItemStorage if the interface expresses the format used for storage/serialization in JSON).

Reply:
You can go with the EthSafeOperation then